components/pagination/src/page-controls.js

```diff
@@ -37,7 +38,7 @@ const PageControls = ({
             secondary
             className="button-next"
             small
-            disabled={page === pageCount}
+            disabled={isLastPage || page === pageCount}
```

Comment:
(minor) could this move up to the parent component so we don't need to deal with this branching logic (and additional component) here?  Basically, this component would only have a boolean `isLastPage` prop and the caller would pass `isLastPage={isLastPage || page === pageCount}`